assets/javascripts/discourse/components/tc-message.js

```diff
@@ -444,4 +445,31 @@ export default Component.extend({
       this.onSelectMessage(this.message);
     });
   },
+
+  @action
+  copyLinkToMessage() {
+    this.element
+      .querySelector(".link-to-message-btn")
+      ?.classList?.add("copied");
+
+    const { protocol, host } = window.location;
+    const endpoint = `/chat/channel/${this.details.chat_channel_id}/chat?messageId=${this.message.id}`;
+    const url = `${protocol}//${host}${getURL(endpoint)}`;
```

Comment:
Do you really need protocol and host here? I thought `getURL(endpoint)` would be enough

Reply:
yes right, it needs better handling in some cases